### PR TITLE
typecheck: Make sure visit_compare works for 'in'

### PR DESCRIPTION
**Pull request by simeonkr**

### CODE REVIEW

File: tests/test_type_inference/test_compare.py [hunk 1]
```diff
@@ -23,16 +22,10 @@ def test_compare_equality(left_value, operator_value_tuples):
     assert compare_node.inf_type.getValue() == bool
 
 
-@nottest
-@given(hs.lists(cs.comparator_operator, min_size=3), cs.homogeneous_list(min_size=4))
+@given(hs.lists(cs.comparator_operator, min_size=3), cs.numeric_list(min_size=4))
```
Comment (simeonkr) [inline tests/test_type_inference/test_compare.py]: Test previously failed due to attempt to apply '<' and '>' to booleans.

File: python_ta/typecheck/typeshed/builtins.pyi [hunk 2]
```diff
@@ -297,7 +297,7 @@ class str(Sequence[str]):
     def __ge__(self, x: str) -> bool: ...
 
     def __len__(self) -> int: ...
-    def __contains__(self, s: object) -> bool: ...
+    def __contains__(self, s: str) -> bool: ...
```
Comment (david-yz-liu) [inline python_ta/typecheck/typeshed/builtins.pyi]: Hmm this is interesting. In principle this file should be an exact copy of `https://github.com/python/typeshed/blob/master/stdlib/3/builtins.pyi`, so I'm hesitant to change it. On the other hand, it does seem like any non-string type is an error. Can you please create a `README.md` under typeshed to make a note of the changes to this file?

File: python_ta/typecheck/typeshed/builtins.pyi [hunk 3]
```diff
@@ -391,7 +391,7 @@ class bytes(ByteString):
     def __rmul__(self, n: int) -> bytes: ...
     if sys.version_info >= (3, 5):
         def __mod__(self, value: Any) -> bytes: ...
-    def __contains__(self, o: object) -> bool: ...
+    def __contains__(self, o: bytes) -> bool: ...
```
Comment (david-yz-liu) [inline python_ta/typecheck/typeshed/builtins.pyi]: On the other hand, this (and likely the others down below) is too strict:

```python
>>> 3 in b'123'
False
```
I don't mind making the type-checker more strict here, but you should articulate this in the README as well.

File: python_ta/typecheck/typeshed/builtins.pyi [hunk 4]
```diff
@@ -549,7 +549,7 @@ class slice:
 class tuple(Sequence[_T_co], Generic[_T_co]):
     def __init__(self, iterable: Iterable[_T_co] = ...) -> None: ...
     def __len__(self) -> int: ...
-    def __contains__(self, x: object) -> bool: ...
+    def __contains__(self, x: _T_co) -> bool: ...
```
Comment (david-yz-liu) [inline python_ta/typecheck/typeshed/builtins.pyi]: Please don't change this one. I want to treat tuples differently, later.

File: tests/test_type_inference/test_compare.py (hunk 1 above)
Comment (david-yz-liu) [reply tests/test_type_inference/test_compare.py]: This is a nice observation, thanks. Although I guess `bool` inherits from `int`, so eventually we'll allow this. :)

**Pull request merged**